dataprofiler/labelers/base_model.py

```diff
@@ -5,22 +5,24 @@
 import copy
 import inspect
 import warnings
-from typing import Any, Dict, List, Optional, Tuple, Type, Union, cast
+from typing import Any, Dict, List, Optional, Tuple, Type, TypeVar, Union, cast
 
 from dataprofiler._typing import DataArray
 
+T = TypeVar("T", bound="AutoSubRegistrationMeta")
+
 
 class AutoSubRegistrationMeta(abc.ABCMeta):
     """For registering subclasses."""
 
     def __new__(
-        cls, clsname: str, bases: Tuple[type, ...], attrs: Dict[str, object]
-    ) -> AutoSubRegistrationMeta:
+        cls: Type[T], clsname: str, bases: Tuple[type, ...], attrs: Dict[str, object]
+    ) -> T:
         """Create auto registration object and return new class."""
-        new_class: Any = super(AutoSubRegistrationMeta, cls).__new__(
+        new_class: T = super(AutoSubRegistrationMeta, cls).__new__(
             cls, clsname, bases, attrs
         )
-        new_class._register_subclass()
+        new_class._register_subclass()  # type: ignore
```

Comment:
I'm not 100% positive, but I think the bound above should be `BaseModel` which would fix this.

We should really try to avoid using these bandaid fixes with `#type: ignore` unless absolutely necessary

Reply:
@Sanketh7 can we get rid of this `#type: ignore`?

Reply:
Changing the typevar to `BaseModel` didn't end up fixing it because mypy doesn't detect the relationship between `BaseModel` and `AutoSubRegistrationMeta`.

Reply:
This has some interesting suggestions:
https://stackoverflow.com/questions/66121127/calling-new-on-an-any

However, if we can't do this, ultimately we should cast referring to the output as a derivation of BaseModel of some type. Returning it as `AutoSubRegistrationMeta` is not truly what matter since it is just a mixin.

Reply:
For now, @taylorfturner maybe we need to do an ignore, but we should validate that when using a model elsewhere in code, it is not consider a AutoReg by type

Reply:
Yeah, I lean toward just making sure elsewhere in the code models that inherit `BaseModel` are validated such that we ensure models have subclass of `BaseModel` @JGSweets @Sanketh7

Reply:
To be clear, I think that should go in a separate PR if we have to add anything to further validate. For now, I think as implemented is okay with regards to this PR getting merged....................

Reply:
+1

Reply:
I'd recommend in order to get the rest of this merged, @Sanketh7, revert this change for the `AutoSubRegistration` and get it working to pass `mypy` checks. Then in a follow-up PR, we can resolve this specific issue more fully.

Reply:
Reverted these changes so we can tackle it in another PR.